From d47eac54afab000b8b273d6a7e7dbbcf4764ab5d Mon Sep 17 00:00:00 2001
From: Surge1223 <surge1223@gmail.com>
Date: Wed, 22 Feb 2017 20:45:04 -0600
Subject: [PATCH 07/13] sepolicy: allow masquerade to read and write theme
 assets

Fix for masquerade to handle theme assets including fonts and bootanimation, also
takes into account when /data/system/theme doesnt exist

SELinux : avc: denied { find } for service=display pid=10652 uid=1000 scontext=u:r:masquerade:s0
tcontext=u:object_r:display_service:s0 tclass=service_manager permissive=0

Change-Id: I1136f9965a85d9c9f19827200e61e4d1c9724593
---
 masquerade.te | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/masquerade.te b/masquerade.te
index 949699c..2f17030 100644
--- a/masquerade.te
+++ b/masquerade.te
@@ -49,3 +49,15 @@ allow masquerade kernel:system module_request;
 allow masquerade dalvikcache_data_file:file { setattr write };
 allow masquerade system_data_file:dir { add_name create rmdir remove_name setattr write };
 allow masquerade system_data_file:file { create getattr setattr unlink write read };
+
+# Allow handling of theme assets
+allow masquerade theme_data_file:dir { add_name create remove_name setattr write rmdir };
+allow masquerade theme_data_file:file { create setattr write unlink };
+
+# Modify system properties
+allow masquerade init:unix_stream_socket connectto;
+allow masquerade mount_service:service_manager find;
+allow masquerade property_socket:sock_file write;
+allow masquerade system_prop:property_service set;
+allow masquerade connectivity_service:service_manager find;
+allow masquerade display_service:service_manager find;
-- 
2.9.4

